homework_02/Martin_Georgiev_13/views.py

```diff
@@ -5,19 +5,67 @@
 
 def trains(request):
 
-    cur_time = datetime.datetime.now(pytz.timezone('America/New_York'))
-    response = requests.get("https://api-v3.mbta.com/predictions?page%5Boffset%5D=0&page%5Blimit%5D=10&sort=-departure_time&filter%5Bstop%5D=place-north").json()
+    params = {
+        "sort" : "-arrival_time",
+        "filter[direction_id]" : "0",
+        "page[offset]" : "0",
+        "include" : "trip,schedule",
+        "page[limit]" : "100",
+        "filter[route_type]" : "2",
+        "filter[stop]" : "place-north",
+        "filter[direction_id]" : "0",
+    }
+
+    current_time = datetime.datetime.now(pytz.timezone("America/New_York"))
+    response = requests.get("https://api-v3.mbta.com/predictions", params = params).json()
+
+    destination_times = []
+    trip_ids = []
+    directions = []
     statuses = []
-    for status in response['data']:
-        statuses.append(status['attributes']['status'])
+    train_ids = []
+
+    for trip in response["data"]:
+        trip_ids.append(trip["relationships"]["trip"]["data"]["id"])
+
+    for time in response["included"]:
+        if time["type"] == "schedule":
+            for id in trip_ids:
+                if time["relationships"]["trip"]["data"]["id"] == id:
+                    timezone = pytz.timezone("America/New_York")
+                    dest_time = datetime.datetime.strptime(datetime.datetime.fromisoformat(str(time["attributes"]["departure_time"])).strftime("%Y %m %d %I:%M %p"), "%Y %m %d %I:%M %p")
+                    if current_time < timezone.localize(dest_time):
+                        destination_times.append(datetime.datetime.fromisoformat(str(time["attributes"]["departure_time"])).strftime("%I:%M %p"))
+                    else:
+                        trip_ids.remove(id)
+
+    for train in response["included"]:
+        if train["type"] == "trip":
+            for id in trip_ids:
+                if train["id"] == id:
+                    train_ids.append(train["attributes"]["name"])
+
+    for direction in response["included"]:
+        if direction["type"] == "trip":
+            for id in trip_ids:
+                if direction["id"] == id:
+                    directions.append(direction["attributes"]["headsign"])
+
+    for status in response["data"]:
+        statuses.append(status["attributes"]["status"])
```

Comment:
The algorithm seems perfect for me. You have taken the right attributes. You can simplify some of the loops with lambda expressions, but I think here is okay without lambda expressions, too.